lib/api/core/models/repositories/repository.js

```diff
@@ -334,24 +338,28 @@ Repository.prototype.persistToCache = function (object, opts) {
   var
     options = opts || {},
     key = options.key || this.index + '/' + this.collection + '/' + object._id,
-    ttl = this.ttl,
-    doThen = response => {
-      if (response === 'OK') {
-        return q(object);
-      }
-      return q.reject(response);
-    };
+    ttl = this.ttl;
 
   if (options.ttl !== undefined) {
     ttl = options.ttl;
   }
   if (ttl === false) {
     return this.cacheEngine.set(key, JSON.stringify(this.serializeToCache(object)))
-      .then(doThen);
+      .then(() => {
+        return object;
+      })
+      .catch(error => {
+        return error;
```

Comment:
Is it intended to resolve the promise with an error as its resolved value, instead of rejecting it?